credentials/alts/internal/conn/record.go

```diff
@@ -120,16 +121,18 @@ func NewConn(c net.Conn, side core.Side, recordProtocol string, key []byte, prot
 		// altsRecordDefaultLength (bytes) data into protected at one
 		// time. Therefore, 2*altsRecordDefaultLength-1 is large enough
 		// to buffer data read from the network.
-		protected = make([]byte, 0, 2*altsRecordDefaultLength-1)
+		protectedBuf = make([]byte, 0, 2*altsRecordDefaultLength-1)
+	} else {
+		protectedBuf = make([]byte, len(protected))
```

Comment:
This doesn't copy protected.

Is there an easy way to add a unit test?

Reply:
Not sure how I missed copying. This is the main reason behind this PR :)

We need a unit test for the case where `protected` is not nil. I'm working on adding that but it would require some setting up. We can add it in this PR or in a follow up one, either is fine with me.

Reply:
Done. I added a test with a non-nil protected buffer. PTAL.